### ISSUE TITLE
Type of event

### ISSUE THREAD

**Issue by mckillo** [jemproject/JEM-Project#1603]
JEM has categories, but doesn't have event types as second level. For example in school, I can categories tree with stages, courses, classrooms, subjects, ... and the each teacher can create event for his subjects. All events have the same level in this category, he can't differentiate by event type, for example: Homework, tasks, activity, workshop, excursion, exam, info, party, theater, sports, question, test, presentation,  ....
I'm sure, that in other sector, we need to have event types, because categories is for organice by venues, countries, companies, forum, building, ....

Proposal, create an event types list in backend and select of event type in the setting of event,as category.
What do you think?

EDIT: typos

**Comment by jojo12** [comment]
I don't understand, sorry

**Comment by Heklaterriol** [comment]
Maybe we could find a way to use the label feature of Joomla for events …

**Comment by Heklaterriol** [comment]
I just checked: it is possible to assign tags to articles, contact and categories. So I think it should be possible to assign them to JEM events too …

**Comment by Heklaterriol** [comment]
see https://docs.joomla.org/Tags_API_Guide

**Comment by mckillo** [comment]
Yes, I like it, it can Bros the way to do it. Let's study and test it. Thanks.

**Comment by mckillo** [comment]
I have to evaluate use tags or not, in the initial review, there is very few api doc about to it, because I think that it's complex and slow. I want to see others extensions if they are used tags.

**Comment by Heklaterriol** [comment]
Thank you.
I think it should be possible to have freely definable labels.
I recenty had a client, who organized a festival and needed very specific lables/categories.

**Comment by mckillo** [comment]
That is the idea, define the type of event, for example the categories can be company and type can be meeting, comercial, marketing. In a school, the categories can be course and letter (3ºB) and type event can be 'Homework, tasks, activity, workshop, excursion, exam, info, party,,..".

And onve it's implemented, new microdata needs to be added:
`<span itemprop="additionalType">type_defined</span>`

In the demo, can be loaded with a types list for event, category and venue.

**Comment by mckillo** [comment]
First, I have to finish the design of the model on paper. I have created a new branch for it: types-feature. I will upload the proposal of this issue. 

If there is no objection, I would like to include this in version 4.4.0, along with the other change in the modules, which is already 
partly in the 'layout-in-modules' branch.

As proposal time ... the date for 4.4.0 could be at the beginning of 2025.

**Comment by reilldesign** [comment]
@mckillo Keyword tagging is a great addition to events. Are these then selected in the event itself and written to the event database table?

**Comment by mckillo** [comment]
<img width="2320" height="800" alt="Image" src="https://github.com/user-attachments/assets/7b378e6c-c606-4cb4-b7d3-ecf6389f291d" />

This new feature commit has been included in: https://github.com/jemproject/JEM-Project/tree/JEM-4.5.0-dev-bugfix-newfeatures. To use it, please make sure you have the latest JEM 4.5.0 version installed. See #2041 for more details about the new release.

**Comment by jojo12** [comment]
on fresh install when clicking load sampledate: 1136 Column count doesn't match value count at row 1

**Comment by mckillo** [comment]
You have to run part os 4.5 0 sql, it's for type.

**Comment by mckillo** [comment]
There are a new sample data .sql . See #2041 (last commitS)

**Comment by mckillo** [comment]
Replace the Font Awesome library selector with an icon category filter. more options to the admin.

<img width="2507" height="995" alt="Image" src="https://github.com/user-attachments/assets/f402daad-d339-499b-8f52-63c2fd68025a" />

**Comment by mckillo** [comment]
<img width="1023" height="1244" alt="Image" src="https://github.com/user-attachments/assets/756b2c1b-d755-41d5-b308-c3eb7ce3c32d" />

Make type badges visually distinct from event status and availability badges by styling them as rounded pill tags.

Reduce the type badge font size and adjust padding so the badge feels secondary to event titles while keeping the label readable and balanced.

**Comment by mckillo** [comment]
I have added language translation options to the type admin view.
The type changes according to the user's language translation.

<img width="1830" height="960" alt="Image" src="https://github.com/user-attachments/assets/b8674f89-b6bd-4843-b424-add2ac579553" />

Name and description are saved as JSON with the different languages into translations filed of _jem_type table.

`{"es-ES":{"name":"Fiesta","description":"Eventos con formato fiesta. Y si terminas tarde acabas tomando churros con chocolate."}}`

Everything before the first period in the first sentence is used as the tooltip title for the type badge.

<img width="774" height="101" alt="Image" src="https://github.com/user-attachments/assets/6530b4d5-cbc5-4d41-8954-53edf86bd10a" />

**Comment by mckillo** [comment]
Included in JEM 4.5.0. beta 2

**Comment by mckillo** [comment]
Sample of types for an English academy

<img width="1396" height="797" alt="Image" src="https://github.com/user-attachments/assets/a7c567a4-a565-470e-812f-472bebe4cd49" />
